src/connection.ts

```diff
@@ -119,8 +122,12 @@ export class Connection extends JSForceConnection {
 
     const conn = new this(options);
     await conn.init();
-    if (!versionFromConfig) {
-      await conn.useLatestApiVersion();
+    if (!versionFromConfig && conn.options?.connectionOptions?.instanceUrl) {
+      try {
+        await conn.useLatestApiVersion();
+      } catch (e) {
+        conn.logger.error(`Cannot connect to the org at ${conn.options?.connectionOptions?.instanceUrl}`);
+      }
```

Comment:
useLatestApiVersion also has a try/catch and it does not rethrow, so this try/catch seems unnecessary.

Reply:
Did some updates after Steve's feedback and a lot more e2e testing.  That try/catch in `useLatestApiVersion` is now more error-specific for DNS problems.

Reply:
Not sure what other scenario would prevent us from being able to determine the version, though.